### ISSUE TITLE
A11y addon ignores inherited parameters

### ISSUE THREAD

**Issue by Hypnosphi** [storybookjs/storybook#11489]
**Describe the bug**
A11y addon only takes into account own story parameters, not the inherited ones

**To Reproduce**
- Apply following patch:
```patch
<+>UTF-8
===================================================================
--- examples/official-storybook/preview.js	(revision 4069ca2e9eccd7e3967bf963450710bd0d7ed5d7)
+++ examples/official-storybook/preview.js	(date 1594337200567)
@@ -145,6 +145,9 @@
   a11y: {
     config: {},
     options: {
+      rules: {
+        'color-contrast': { enabled: false },
+      },
       checks: { 'color-contrast': { options: { noScroll: true } } },
       restoreScroll: true,
     },

```
- run `yarn start`
- go to `http://localhost:9011/?path=/story/ui-panel--default`, switch to accessibility tab

**Expected behavior**
No violations reported, because `color-contrast` rule should be disabled.

It works correctly if I define the parameter on story level, but doesn't if I do it on kind or global level

**Comment by tmeasday** [comment]
@tooppaaa - I dug into this issue a bit and the immediate bug is due to this line:

https://github.com/storybookjs/storybook/blob/badece61154f4cd372404786186125a7369e362f/addons/a11y/src/a11yRunner.ts#L46

Accessing both the variable on `window`, and specifically the private member (`_stories`) is not a great idea! 

The simplest fix is to just use `fromId`: https://github.com/storybookjs/storybook/tree/11489-fix-parameters-in-a11y -- this does resolve @Hypnosphi's issue but I took at look at the PR that change came from: https://github.com/storybookjs/storybook/pull/10381/files and I'm sort of wondering what's going on here in the first place.

It seems like we are getting the parameters in this non-standard "anti-pattern" kind of way because we are avoiding using a decorator (the PR itself says "issue: a11y use decorator"). I'm not sure what the rationale for that is? Is it to avoid the user having to add a decorator?

If so, it would be better to just add the decorator directly by exporting it on the `decorators` key from one of the `config` entries you've added (you could export directly from that file, `a11yRunner.ts` for instance). It seems like the previous approach of using a decorator to keep track of the current story, but only running axe when receiving an event was more idiomatic and "correct".

Is there something I'm missing here?

**Comment by shilman** [comment]
Ooh-la-la!! I just released https://github.com/storybookjs/storybook/releases/tag/v6.0.0-rc.22 containing PR #11730 that references this issue. Upgrade today to try it out!

You can find this prerelease on the `@next` NPM tag.
